package.json

```
@@ -137,8 +138,8 @@
   "dependencies": {
     "@carbon/icon-helpers": "10.37.0",
     "@carbon/icons": "11.14.0",
-    "@carbon/telemetry": "0.1.0",
     "@carbon/utils-position": "1.1.4",
+    "@ibm/telemetry-js": "^1.2.1",
```

```suggestion
    "@ibm/telemetry-js": "1.2.1",
```

Can we version lock this package? This is more of a safe measure in case there is a breaking change made without a major version bump.  Will need to run `npm install` to apply these changes to package-lock as well.

---

Hey, @Akshat55! You're certainly more than welcome to pin the IBM Telemetry version to a specific release if you'd like. I will say though that by design, we (the IBM Telemetry team) enforce that our code be zero-impact to any library instrumented with it. If our code breaks or fails, our fundamental design principle is to fail silently. We will not impact any build/CI/install processes under any circumstances even if we encounter errors when running. We accomplish this by running our data collection in a detached process which (regardless of its exit code) will always be silent. In the event that we do have a breaking change (maybe a config file format change or a collection regression bug), that will not impact you or your library's consumers at all.

Looking at your `package.json` file, it looks like your team's convention is to pin all deps, and if this is your preferred approach, it's totally fine by us. I just wanted to offer some peace of mind about how the telemetry tooling operates.

If you'd like @francinelucca or I to pin the dep, just let us know and we can make that adjustment, or if you'd like to make that change yourself, that's fine too!

---

Thank you for the clarification @jdharvey-ibm!